packages/langium/src/syntax-tree.ts

```diff
@@ -197,19 +197,24 @@ export abstract class AbstractAstReflection implements AstReflection {
 export interface TypeMetaData {
     /** The name of this meta model type. Corresponds to the `AstNode.$type` value. */
     name: string
-    /** A list of mandatory properties. These are defaults for array and boolean based properties (`[]` and `false` respectively). */
-    mandatory: TypeMandatoryProperty[]
+    /** A list of mandatory properties. They contain default values for their respective property in the AST. */
+    mandatory: MandatoryProperty[]
```

Comment:
This is a breaking change. Can we keep the old type and mark it deprecated? Otherwise we should wait for v3.0.0 before merging this.
```suggestion
    mandatory: Array<MandatoryProperty | TypeMandatoryProperty>
```

Reply:
We decided to move the feature to v3.0.0, so the change is ok.